Core/clim-core/menu.lisp

```diff
@@ -374,35 +373,34 @@ account, and create a list of menu buttons."
 (defmethod disarm-menu ((object menu-bar))
   (setf (slot-value object 'armed) nil))
 
-(defun %make-menu-contents (command-table)
+(defun %make-menu-contents (command-table frame)
   (with-slots (menu) (find-command-table command-table)
-    (append
-     (loop for item in menu
-	collect
-	  (make-menu-button-from-menu-item
-	   item nil
-	   :bottomp t
-	   :vertical nil
-	   :command-table command-table))
-     (list +fill+))))
+    (append (loop for item in menu
+                  collect (make-menu-button-from-menu-item
+                           item nil frame
+                           :bottomp t
+                           :vertical nil
+                           :command-table command-table))
+            (list +fill+))))
 
 (defun make-menu-bar (command-table
-		      &key width height
-		           (max-width +fill+) max-height
-			   min-width min-height)
-  (with-slots (menu) (find-command-table command-table)
-    (make-pane-1 *pane-realizer* *application-frame*
-		 'menu-bar
-		 :background *3d-normal-color*
-		 :width width :height height
-		 :max-width max-width :max-height max-height
-		 :min-width min-width :min-height min-height
-		 :contents (%make-menu-contents command-table))))
+                      &key width height
+                           (max-width +fill+) max-height
+                           min-width min-height)
+  (let ((frame *application-frame*))
+    (with-slots (menu) (find-command-table command-table)
+      (make-pane-1 *pane-realizer* frame
+                   'menu-bar
+                   :background *3d-normal-color*
+                   :width width :height height
+                   :max-width max-width :max-height max-height
+                   :min-width min-width :min-height min-height
+                   :contents (%make-menu-contents command-table frame)))))
```

Comment:
First two done.

Regarding passing the menu bar: The client argument is `nil` in some of the calls so merging the `client` and `frame` parameters by passing the menu bar is at least not trivially possible. If that's fine with you, I would prefer to postpone this enhancement.

Reply:
The only call with client being NIL is (afaik) from `%make-menu-contents` and that would be addressed with a change I have mentioned.

Reply:
Some highlights:

`contents` is an argument in `panes` to initialize-instance:

```
(defmethod initialize-instance :after ((pane box-pane) &key contents)
  (setf (%pane-contents pane) contents))

;; below gf has typo in a name and is apparently not used.
(defgeneric %pane-contests (pane contents))

;; and lengy setf method follows which is indeed used for box-panes
```

also argument `:client` is argument initializing slot returned by a reader gadget-client. While we pass it as nill from make-menu-button-from-menu-item it is most likely because we create contents during creation of menu-bar, so we can't pass the object which is being created. This is fixed later in an after method defined in menu.lisp file:

```
(defmethod (setf %pane-contents) :after (contents (pane menu-bar))
  (declare (ignore contents))
  (setf (slot-value pane 'items) (copy-list (sheet-children pane)))
  (loop for child in (menu-children pane)
        do (setf (gadget-client child) pane)))
```